### ISSUE TITLE
Modal focus management

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3414]
The modal can be improved with better focus management to help keyboard users to access content more logically.

* [ ] When opening a modal, focus should always move to the modal title (also with wizards)
* [ ] When closing a modal, focus should always return to the previous location

This should already be happening with our focus trap, but it was reported to have some problems with wizards so perhaps there is a problem only in how wizards are using it. Need to verify the behavior fully in modals and wizards.

**Comment by coryrylan** [comment]
I replicated the issue. The focus trap works in Chrome but does not work in Safari. Going to do some more browser testing as well.

**Comment by chlane** [comment]
There are no modal trapping issues reported at http://cx-guidelines.vmware.com/clarityTesting/. 

But Cory and I did discuss issue #117 at http://cx-guidelines.vmware.com/clarityTesting/ about screen readers being able read content outside of modal dialogs. Due to the lack of support for aria-modal, 
 the ideal solution is to hide the rest of the application from assistive technologies using aria-hidden="true" while the modal is in view but this is probably not possible unless the the modal is a sibling element to the rest of the application. 

So the best we can do is:
1) use aria-modal="true" on the dialog and wait for support to improve
2) add textual boundaries i.e, "beginning of edit dialog" "end of edit dialog" to tell the user if that they are reading a dialog and where its boundaries are.
